www/tsconfig.json

```diff
@@ -1,35 +1,14 @@
 {
+  "extends": "../tsconfig.json",
   "compilerOptions": {
-    "disableSourceOfProjectReferenceRedirect": true,
-    "disableSolutionSearching": true,
-    "target": "es6",
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "es6"
-    ],
-    "jsx": "react",
-    "module": "es6",
-    "moduleResolution": "node",
-    "preserveSymlinks": true,
-    "allowJs": true,
-    "skipLibCheck": true,
-    "esModuleInterop": true,
-    "allowSyntheticDefaultImports": true,
-    "strict": true,
-    "forceConsistentCasingInFileNames": true,
-    "noFallthroughCasesInSwitch": true,
+    "rootDir": "src",
     "resolveJsonModule": true,
-    "isolatedModules": true,
-    "noEmit": false,
-    "baseUrl": "src",
+    "noImplicitAny": false,
```

Comment:
Sooo this is not good and should be definitely set to `true` but we cannot do it now. Reason is that core Paragon `index.js` file (the file that exports all component) does not have any typings which results in an error when we try to import something from it into TypeScript file, e.g. if we do `import { Hyperlink } from '~paragon-react';`) typescript does not find any types for `Hyperlink` and thinks it has an implicit `any` type.

We probably need to provide types for `index.js` file, maybe in the follow-up PR? also, we might want to eventually publish our types as a separate package under `@types/paragon`, I think a lot of projects do this when they have typescript, (e.g. [react-bootstrap's types package](https://www.npmjs.com/package/@types/react-bootstrap))

Reply:
@viktorrusakov This types issue came up recently in conversation with @marlonkeating who is trying to bring TypeScript into `frontend-app-admin-portal` where TypeScript was complaining that Paragon's `Button` component (and others) didn't have types defined and would need to define them locally in the application to get around it.

It looks like publishing `@types/paragon` might even be possible to do self-service, according to their docs: 

* https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request
* https://devblogs.microsoft.com/typescript/changes-to-how-we-manage-definitelytyped/#what-pull-requests-can-be-merged-by-definition-owners
* https://github.com/DefinitelyTyped/dt-mergebot/

Based on these readings, it looks like an initial PR would need to be reviewed by the DT maintainers but subsequent edits might not (self-serve), which is cool.

Given this, yeah, I agree with your proposal here... get types defined for `index.js` as a follow-up PR and then try to get `@types/paragon` published. 

I filed two follow-up issues related to this:
* https://github.com/openedx/paragon/issues/1964
* https://github.com/openedx/paragon/issues/1965

Reply:
I suppose the other short term fix for consumers' type issues would be change `noImplicitAny` to `false` in the root `tsconfig.json` temporarily as well.

I'd probably lean towards introducing types vs. changing `noImplicitAny=false` temporarily, though curious to hear what you think!

Reply:
Follow-up from that last comment, it might actually be worth changing `noImplicitAny=false` for now as the quickest path forward. 

This `tsconfig.json` is largely acting as if its a consuming application (e.g., an MFE). When we introduce `tsconfig.json` into `@edx/frontend-build` ([PR](https://github.com/openedx/frontend-build/pull/294)), it's likely we'll want to keep `noImplicitAny=false` in that repo as we don't want the default for MFEs to be strictly-typed since that would require a large effort to refactor everything to TS, so leaving it as optionally-typed supports a more incremental, opt-in migration for MFEs.

That said, we could also decide for the Paragon documentation website that we _do_ want to be strictly-typed as you're proposing here; though, yes, in that case, we would need to define types for all the exports in `index.js`.